### ISSUE TITLE
Report page shows nothing

### ISSUE THREAD

**Issue by trodrigues** [th0r/webpack-bundle-analyzer#10]
Hi there,

I was trying this out by running the two CLI commands mentioned at the end of the readme, and I end up with a generated static report.html that shows just a blank page. There's not even any console or network errors. The code is there in the file, it just doesn't seem to do anything at all.

The stats.json file seems to be generated correctly and with no errors.

I tried this on a couple of separate apps, and also opening the report.html on Chrome, Safari and Firefox.

I can provide the stats.json and generated report.html if desired to help debug the issue but I'd rather not do it directly here.

Alternatively, if there's something I can do myself to help debug it, let me know.

In any case, this seems like an amazing and very helpful idea, so thanks for taking the time to work on this :)

**Comment by th0r** [comment]
Yes, it would be nice if you could send me `stats.json`. Maybe by email? It's `grunin.ya@ya.ru`.
And what command do you use to generate report?

**Comment by th0r** [comment]
@trodrigues Do you still have this bug with latest version?

**Comment by yoda** [comment]
yeah I seem to, it seems to have something to do with chunking and the way the modules are designated in the stats.json. Structure doesn't seem to match the expected structure. As the datapoints are not generated and not put into the data array.

Providing the json schema of the stats.json from our app. not sure if it will be useful.

```js{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "description": "Generated from public/assets/stats.json with shasum a6945465b51461b684dcfa8803f5be130b82bb21",
  "type": "object",
  "required": [
    "errors",
    "warnings",
    "version",
    "hash",
    "publicPath",
    "assetsByChunkName",
    "assets",
    "chunks",
    "modules",
    "filteredModules",
    "children"
  ],
  "properties": {
    "errors": {
      "type": "array",
      "minItems": 0,
      "uniqueItems": true,
      "items": {
      }
    },
    "warnings": {
      "type": "array",
      "minItems": 0,
      "uniqueItems": true,
      "items": {
      }
    },
    "version": {
      "type": "string"
    },
    "hash": {
      "type": "string"
    },
    "publicPath": {
      "type": "string"
    },
    "assetsByChunkName": {
      "type": "object",
      "required": [
        "boot",
        "app",
        "common",
        "reports",
        "images",
        "login",
        "map",
        "audits",
        "mail",
        "datagrid",
        "ie"
      ],
      "properties": {
        "boot": {
          "type": "string"
        },
        "app": {
          "type": "string"
        },
        "common": {
          "type": "string"
        },
        "reports": {
          "type": "string"
        },
        "images": {
          "type": "string"
        },
        "login": {
          "type": "string"
        },
        "map": {
          "type": "string"
        },
        "audits": {
          "type": "string"
        },
        "mail": {
          "type": "array",
          "minItems": 1,
          "uniqueItems": true,
          "items": {
            "type": "string"
          }
        },
        "datagrid": {
          "type": "string"
        },
        "ie": {
          "type": "string"
        }
      }
    },
    "assets": {
      "type": "array",
      "minItems": 1,
      "uniqueItems": true,
      "items": {
        "type": "object",
        "required": [
          "name",
          "size",
          "chunks",
          "chunkNames"
        ],
        "properties": {
          "name": {
            "type": "string"
          },
          "size": {
            "type": "integer"
          },
          "chunks": {
            "type": "array",
            "minItems": 0,
            "uniqueItems": true,
            "items": {
            }
          },
          "chunkNames": {
            "type": "array",
            "minItems": 0,
            "uniqueItems": true,
            "items": {
            }
          }
        }
      }
    },
    "chunks": {
      "type": "array",
      "minItems": 1,
      "uniqueItems": true,
      "items": {
        "type": "object",
        "required": [
          "id",
          "rendered",
          "initial",
          "size",
          "names",
          "files",
          "hash",
          "parents",
          "modules",
          "filteredModules",
          "origins"
        ],
        "properties": {
          "id": {
            "type": "integer"
          },
          "rendered": {
            "type": "boolean"
          },
          "initial": {
            "type": "boolean"
          },
          "entry": {
            "type": "boolean"
          },
          "extraAsync": {
            "type": "boolean"
          },
          "size": {
            "type": "integer"
          },
          "names": {
            "type": "array",
            "minItems": 1,
            "uniqueItems": true,
            "items": {
              "type": "string"
            }
          },
          "files": {
            "type": "array",
            "minItems": 1,
            "uniqueItems": true,
            "items": {
              "type": "string"
            }
          },
          "hash": {
            "type": "string"
          },
          "parents": {
            "type": "array",
            "minItems": 0,
            "uniqueItems": true,
            "items": {
            }
          },
          "modules": {
            "type": "array",
            "minItems": 1,
            "uniqueItems": true,
            "items": {
              "type": "object",
              "required": [
                "id",
                "identifier",
                "name",
                "size",
                "chunks",
                "assets",
                "errors",
                "warnings",
                "reasons"
              ],
              "properties": {
                "id": {
                  "type": "integer"
                },
                "identifier": {
                  "type": "string"
                },
                "name": {
                  "type": "string"
                },
                "index": {
                  "type": "integer"
                },
                "index2": {
                  "type": "integer"
                },
                "size": {
                  "type": "integer"
                },
                "cacheable": {
                  "type": "boolean"
                },
                "built": {
                  "type": "boolean"
                },
                "optional": {
                  "type": "boolean"
                },
                "prefetched": {
                  "type": "boolean"
                },
                "chunks": {
                  "type": "array",
                  "minItems": 1,
                  "uniqueItems": true,
                  "items": {
                    "type": "integer"
                  }
                },
                "assets": {
                  "type": "array",
                  "minItems": 0,
                  "uniqueItems": true,
                  "items": {
                  }
                },
                "issuer": {
                },
                "profile": {
                  "type": "object",
                  "required": [
                    "factory",
                    "building",
                    "dependencies"
                  ],
                  "properties": {
                    "factory": {
                      "type": "integer"
                    },
                    "building": {
                      "type": "integer"
                    }
                  }
                },
                "failed": {
                  "type": "boolean"
                },
                "errors": {
                  "type": "integer"
                },
                "warnings": {
                  "type": "integer"
                },
                "reasons": {
                  "type": "array",
                  "minItems": 0,
                  "uniqueItems": true,
                  "items": {
                  }
                }
              }
            }
          },
          "filteredModules": {
            "type": "integer"
          },
          "origins": {
            "type": "array",
            "minItems": 1,
            "uniqueItems": true,
            "items": {
              "type": "object",
              "required": [
                "moduleId",
                "module",
                "moduleIdentifier",
                "moduleName",
                "loc",
                "name",
                "reasons"
              ],
              "properties": {
                "moduleId": {
                  "type": "integer"
                },
                "module": {
                  "type": "string"
                },
                "moduleIdentifier": {
                  "type": "string"
                },
                "moduleName": {
                  "type": "string"
                },
                "loc": {
                  "type": "string"
                },
                "name": {
                  "type": "string"
                },
                "reasons": {
                  "type": "array",
                  "minItems": 0,
                  "uniqueItems": true,
                  "items": {
                  }
                }
              }
            }
          }
        }
      }
    },
    "modules": {
      "type": "array",
      "minItems": 1,
      "uniqueItems": true,
      "items": {
        "type": "object",
        "required": [
          "id",
          "identifier",
          "name",
          "size",
          "chunks",
          "assets",
          "errors",
          "warnings",
          "reasons"
        ],
        "properties": {
          "id": {
            "type": "integer"
          },
          "identifier": {
            "type": "string"
          },
          "name": {
            "type": "string"
          },
          "index": {
            "type": "integer"
          },
          "index2": {
            "type": "integer"
          },
          "size": {
            "type": "integer"
          },
          "cacheable": {
            "type": "boolean"
          },
          "built": {
            "type": "boolean"
          },
          "optional": {
            "type": "boolean"
          },
          "prefetched": {
            "type": "boolean"
          },
          "chunks": {
            "type": "array",
            "minItems": 1,
            "uniqueItems": true,
            "items": {
              "type": "integer"
            }
          },
          "assets": {
            "type": "array",
            "minItems": 0,
            "uniqueItems": true,
            "items": {
            }
          },
          "issuer": {
          },
          "profile": {
            "type": "object",
            "required": [
              "factory",
              "building",
              "dependencies"
            ],
            "properties": {
              "factory": {
                "type": "integer"
              },
              "building": {
                "type": "integer"
              }
            }
          },
          "failed": {
            "type": "boolean"
          },
          "errors": {
            "type": "integer"
          },
          "warnings": {
            "type": "integer"
          },
          "reasons": {
            "type": "array",
            "minItems": 0,
            "uniqueItems": true,
            "items": {
            }
          }
        }
      }
    },
    "filteredModules": {
      "type": "integer"
    },
    "children": {
      "type": "array",
      "minItems": 1,
      "uniqueItems": true,
      "items": {
        "type": "object",
        "required": [
          "errors",
          "warnings",
          "publicPath",
          "assetsByChunkName",
          "assets",
          "chunks",
          "modules",
          "filteredModules",
          "children",
          "name"
        ],
        "properties": {
          "errors": {
            "type": "array",
            "minItems": 0,
            "uniqueItems": true,
            "items": {
            }
          },
          "warnings": {
            "type": "array",
            "minItems": 0,
            "uniqueItems": true,
            "items": {
            }
          },
          "publicPath": {
            "type": "string"
          },
          "assetsByChunkName": {
            "type": "object",
            "properties": {
            }
          },
          "assets": {
            "type": "array",
            "minItems": 0,
            "uniqueItems": true,
            "items": {
            }
          },
          "chunks": {
            "type": "array",
            "minItems": 1,
            "uniqueItems": true,
            "items": {
              "type": "object",
              "required": [
                "id",
                "rendered",
                "initial",
                "entry",
                "size",
                "names",
                "files",
                "hash",
                "parents",
                "modules",
                "filteredModules",
                "origins"
              ],
              "properties": {
                "id": {
                  "type": "integer"
                },
                "rendered": {
                  "type": "boolean"
                },
                "initial": {
                  "type": "boolean"
                },
                "entry": {
                  "type": "boolean"
                },
                "extraAsync": {
                  "type": "boolean"
                },
                "size": {
                  "type": "integer"
                },
                "names": {
                  "type": "array",
                  "minItems": 0,
                  "uniqueItems": true,
                  "items": {
                  }
                },
                "files": {
                  "type": "array",
                  "minItems": 1,
                  "uniqueItems": true,
                  "items": {
                    "type": "string"
                  }
                },
                "hash": {
                  "type": "string"
                },
                "parents": {
                  "type": "array",
                  "minItems": 0,
                  "uniqueItems": true,
                  "items": {
                  }
                },
                "modules": {
                  "type": "array",
                  "minItems": 1,
                  "uniqueItems": true,
                  "items": {
                    "type": "object",
                    "required": [
                      "id",
                      "identifier",
                      "name",
                      "index2",
                      "size",
                      "cacheable",
                      "built",
                      "chunks",
                      "assets",
                      "profile",
                      "errors",
                      "warnings",
                      "reasons",
                      "source"
                    ],
                    "properties": {
                      "id": {
                        "type": "integer"
                      },
                      "identifier": {
                        "type": "string"
                      },
                      "name": {
                        "type": "string"
                      },
                      "index": {
                        "type": "integer"
                      },
                      "index2": {
                        "type": "integer"
                      },
                      "size": {
                        "type": "integer"
                      },
                      "cacheable": {
                        "type": "boolean"
                      },
                      "built": {
                        "type": "boolean"
                      },
                      "optional": {
                        "type": "boolean"
                      },
                      "prefetched": {
                        "type": "boolean"
                      },
                      "chunks": {
                        "type": "array",
                        "minItems": 1,
                        "uniqueItems": true,
                        "items": {
                          "type": "integer"
                        }
                      },
                      "assets": {
                        "type": "array",
                        "minItems": 0,
                        "uniqueItems": true,
                        "items": {
                        }
                      },
                      "issuer": {
                      },
                      "profile": {
                        "type": "object",
                        "required": [
                          "factory",
                          "building"
                        ],
                        "properties": {
                          "factory": {
                            "type": "integer"
                          },
                          "building": {
                            "type": "integer"
                          }
                        }
                      },
                      "failed": {
                        "type": "boolean"
                      },
                      "errors": {
                        "type": "integer"
                      },
                      "warnings": {
                        "type": "integer"
                      },
                      "reasons": {
                        "type": "array",
                        "minItems": 0,
                        "uniqueItems": true,
                        "items": {
                        }
                      },
                      "source": {
                        "type": "string"
                      }
                    }
                  }
                },
                "filteredModules": {
                  "type": "integer"
                },
                "origins": {
                  "type": "array",
                  "minItems": 1,
                  "uniqueItems": true,
                  "items": {
                    "type": "object",
                    "required": [
                      "moduleId",
                      "module",
                      "moduleIdentifier",
                      "moduleName",
                      "loc",
                      "reasons"
                    ],
                    "properties": {
                      "moduleId": {
                        "type": "integer"
                      },
                      "module": {
                        "type": "string"
                      },
                      "moduleIdentifier": {
                        "type": "string"
                      },
                      "moduleName": {
                        "type": "string"
                      },
                      "loc": {
                        "type": "string"
                      },
                      "reasons": {
                        "type": "array",
                        "minItems": 0,
                        "uniqueItems": true,
                        "items": {
                        }
                      }
                    }
                  }
                }
              }
            }
          },
          "modules": {
            "type": "array",
            "minItems": 1,
            "uniqueItems": true,
            "items": {
              "type": "object",
              "required": [
                "id",
                "identifier",
                "name",
                "index2",
                "size",
                "cacheable",
                "built",
                "chunks",
                "assets",
                "profile",
                "errors",
                "warnings",
                "reasons",
                "source"
              ],
              "properties": {
                "id": {
                  "type": "integer"
                },
                "identifier": {
                  "type": "string"
                },
                "name": {
                  "type": "string"
                },
                "index": {
                  "type": "integer"
                },
                "index2": {
                  "type": "integer"
                },
                "size": {
                  "type": "integer"
                },
                "cacheable": {
                  "type": "boolean"
                },
                "built": {
                  "type": "boolean"
                },
                "optional": {
                  "type": "boolean"
                },
                "prefetched": {
                  "type": "boolean"
                },
                "chunks": {
                  "type": "array",
                  "minItems": 1,
                  "uniqueItems": true,
                  "items": {
                    "type": "integer"
                  }
                },
                "assets": {
                  "type": "array",
                  "minItems": 0,
                  "uniqueItems": true,
                  "items": {
                  }
                },
                "issuer": {
                },
                "profile": {
                  "type": "object",
                  "required": [
                    "factory",
                    "building"
                  ],
                  "properties": {
                    "factory": {
                      "type": "integer"
                    },
                    "building": {
                      "type": "integer"
                    }
                  }
                },
                "failed": {
                  "type": "boolean"
                },
                "errors": {
                  "type": "integer"
                },
                "warnings": {
                  "type": "integer"
                },
                "reasons": {
                  "type": "array",
                  "minItems": 0,
                  "uniqueItems": true,
                  "items": {
                  }
                },
                "source": {
                  "type": "string"
                }
              }
            }
          },
          "filteredModules": {
            "type": "integer"
          },
          "children": {
            "type": "array",
            "minItems": 0,
            "uniqueItems": true,
            "items": {
            }
          },
          "name": {
            "type": "string"
          }
        }
      }
    }
  }
}

**Comment by th0r** [comment]
@yoda Yes, schema is not too helpful.

I've just released `v1.5.0` with `statsOptions` option for analyzer plugin so you can exclude any sensitive information from generated stats file.
For example you can exclude module sources from it with `source: false` option like that:
```js
new BundleAnalyzerPlugin({
  analyzerMode: 'disabled',
  generateStatsFile: true,
  statsOptions: { source: false }
})
```
Maybe you could publish or send me your clipped stats file?

**Comment by th0r** [comment]
@yoda And what's your OS?

**Comment by yoda** [comment]
Managed to get it working I had been trying to use the --profile --json on webpack to generate the file but it generates a different file than your plugin creates. I also found another stats plugin that was probably interfering.

**Comment by yoda** [comment]
And os is Linux vagrant 3.19.0-25-generic #26~14.04.1-Ubuntu

**Comment by th0r** [comment]
@yoda Nice to hear.
@trodrigues Do you still have this issue?

**Comment by yoda** [comment]
@th0r Really awesome project gunna help me alot! Depressing to see how bad my bundle is, but at least I can see it now.

**Comment by th0r** [comment]
@trodrigues Do you have this issue or I can close it?

**Comment by slestang** [comment]
I have the same issue; empty report.

Also running Ubuntu. My webpack packages:
webpack@1.13.2
webpack-bundle-analyzer@1.5.3
webpack-bundle-size-analyzer@2.2.0

The  main part of my Webpack conf:
```javascript
{
    entry: {
      main: [
        './src/js/app.js',
      ],
      vendors: './src/styles/vendors.scss',
      bootstrap: './src/js/bootstrap.js',
    },
    output: {
      filename: 'js/' + filesPrefix + '[name].js',
      chunkFilename: 'js/' + filesPrefix + '[name].js',
    },
    plugins: [
      new webpack.DefinePlugin({
        __DEV__: isDev,
        __DEVTOOL__: false,
        'process.env': {
          NODE_ENV: JSON.stringify(isDev ? 'development' : 'production'),
        },
      }),
      // we use the ContextReplacementPlugin to import only the english locale from moment.js
      // see http://stackoverflow.com/a/25426019/5788800
      new webpack.ContextReplacementPlugin(/moment[\/\\]locale$/, /en/),
      new webpack.optimize.DedupePlugin(),
      new webpack.optimize.CommonsChunkPlugin({
        name: 'vendors',
        chunks: ['main', 'vendors'],
        filename: 'js/' + filesPrefix + '[name].js',
        // taken from http://stackoverflow.com/a/29087883/5788800
        // the function below put all module which aren't in /src into
        // vendors.js (aka all files from node_modules)
        minChunks: function(module, count) {
          return module.resource && module.resource.indexOf(src) === -1;
        },
      }),
      new webpack.optimize.CommonsChunkPlugin({
        name: 'bootstrap',
        chunks: ['bootstrap'],
        filename: 'js/bootstrap.js',
        minChunks: Infinity,
      }),
      new webpack.ProvidePlugin({
        jQuery: 'jquery',
      }),
      new ExtractTextPlugin('css/' + filesPrefix + '[name].css', {
        allChunks: true,
      }),
      new BundleAnalyzerPlugin(),
    ],
    resolve: {
      root: [
        'src',
      ],
    },
   // some other config
  }
```
The div#treemap contains 3 canvas, which I suppose match my 3 entry points, but they are all blank.

**Comment by th0r** [comment]
@slestang can you provide your webpack stats file?

**Comment by slestang** [comment]
For information, after some debugging by @th0r the issue for me seems to came from trying to use webpack-bundle-analyzer with WebpackDevServer and/or my _dev_ config.
Doing a simple _webpack --config webpack.config.production.js_ generates a non blank report.

I will add that using this plugin on a _dev_ config is certainly not what you want as some pieces of code are removed from the bundles with  _production_ config,  like React's PropTypes, so you can end-up with very different results between _dev_ and _production_.

**Comment by pfrohner** [comment]
I had the same issue, replacing "webpack-stats-plugin" with "stats-webpack-plugin" was the solution in my case...

thanks for this awesome tool!

**Comment by trodrigues** [comment]
Hi, sorry, I'm horrible at keeping up with github notifications.

Yes, I do still have the issue. I'll send you the stats file.

**Comment by trodrigues** [comment]
By the way, by using the configuration in my webpack file I managed to get this to work.

The problem seems to be only when generating the stats file with `webpack --profile --json > stats.json` and then running `webpack-bundle-analyzer stats.json build`.

If I run `webpack-bundle-analyzer` separately with the stats file generated by using the config in the webpack file, it also works fine.

**Comment by th0r** [comment]
@trodrigues Fixed and issue with your stats file in `v2.2.0`

**Comment by trodrigues** [comment]
Great :) Thanks!
